lib/Frontend/CASOutputBackends.cpp

```diff
@@ -57,16 +57,39 @@ class SwiftCASOutputBackend::Implementation {
 public:
   Implementation(ObjectStore &CAS, ActionCache &Cache, ObjectRef BaseKey,
                  const FrontendInputsAndOutputs &InputsAndOutputs,
+                 const FrontendOptions &Opts,
                  FrontendOptions::ActionType Action)
       : CAS(CAS), Cache(Cache), BaseKey(BaseKey),
-        InputsAndOutputs(InputsAndOutputs), Action(Action) {
+        InputsAndOutputs(InputsAndOutputs), Opts(Opts), Action(Action) {
     initBackend(InputsAndOutputs);
   }
 
   llvm::Expected<std::unique_ptr<llvm::vfs::OutputFileImpl>>
   createFileImpl(llvm::StringRef ResolvedPath,
                  std::optional<llvm::vfs::OutputConfig> Config) {
     auto ProducingInput = OutputToInputMap.find(ResolvedPath);
+    if (ProducingInput == OutputToInputMap.end() &&
```

Comment:
Instead of doing this special condition, I prefer add symbol graph to `SupplementaryOutputPaths` and compute the path in `ArgsToFrontendOutputsConverter.cpp`.

Hope it is not too bad to pass the output path to the location where the file is generated.

Reply:
> I prefer add symbol graph to SupplementaryOutputPaths and compute the path in ArgsToFrontendOutputsConverter.cpp.

The symbol graph contains an arbitrary number of files, and I think you can't know how many files there will be until you typecheck, because if you have module A that contains `import B; extension B.Foo { public func bar() {} }` there will be `A.symbols.json` and `A@B.symbols.json`.  That's why I didn't model it as a "normal" output.